components/collector/src/source_collectors/jira/issues.py

```diff
@@ -115,3 +117,12 @@ def __get_sprint_names(cls, sprint_texts: list[str]) -> str:
         matches = [cls.SPRINT_NAME_RE.search(sprint_text) for sprint_text in sprint_texts]
         sprint_names = [match.group(1) for match in matches if match]
         return ", ".join(sorted(sprint_names))
+
+    async def _determine_max_results(self) -> int:
+        """Maximum number of issues to retrieve per page."""
+        if isinstance(self.MAX_RESULTS, int):
+            return self.MAX_RESULTS
```

Comment:
Can't we make this a cached property so that `self.MAX_RESULTS` isn't needed anymore?

Reply:
Well, it's a bit hairy due to the asyncio coroutine :( I originally wanted to fetch it in __init__ but then I need to use a different url library, such as requests. I don't think we have an async `cached_property` version somewhere in collector_utilities, is this desirable?